### ISSUE TITLE
be able to print any type

### ISSUE THREAD

**Issue by acarumut** [MPLLang/mpl#165]
it is a pain to have to write print functions
i have encountered a good example  (simplifying here to make the point)

i was writing a function 
``` 
fun f (x: 'a datastructure): 'a datastructure = 
... 
bunch of code
```

now i am testing with `'a = int` but if i try to print `x` inside `f`, then i break polymorphism because i am assuming int :(
so basically i have to change my type signature etc...

**Comment by shwestrick** [comment]
Yeah this would be fantastic to have. Conceptually it seems straightforward for the compiler to provide a polymorphic primitive `toString: 'a -> string`. We could elaborate it into real code immediately after monomorphization.

Generating pretty printers automatically for arbitrary types seems like it could be a pain! Perhaps we could start with simple things (e.g. just base types, integers, tuples, etc.) and extend it incrementally with support for more interesting stuff.

**Comment by umutacar** [comment]
would it be feasible to extend the compiler with an "anytime" expression that gets passed through the various stages of the compilation? if so, then this and similar things could be easy to implement perhaps?

**Comment by acarumut** [comment]
cool!
